gsa/edits.go

```diff
@@ -15,6 +15,36 @@ package gsa
 //      as a string.
 func GetEdits(p, q string) (gapFreeP, gapFreeQ, edits string) {
 	gapFreeP, gapFreeQ, edits = "", "", ""
+
+	// step 1
+	if p == "" && q == "" {
+		return gapFreeP, gapFreeQ, edits
+	}
+
+	// step 2
+	if p[0:1] != "-" && q[0:1] != "-" {
```

Comment:
I think

```go
if p[0] != '-' && q[0] != '-' { ...
```

would be the traditional way or writing this. You don't need to work with strings if bytes are what you really want to look at.